bundles/org.openhab.binding.volvooncall/src/main/java/org/openhab/binding/volvooncall/internal/action/VolvoOnCallActions.java

```diff
@@ -44,20 +46,20 @@ public VolvoOnCallActions() {
     @Override
     public void setThingHandler(@Nullable ThingHandler handler) {
         if (handler instanceof VehicleHandler) {
-            this.handler = (VehicleHandler) handler;
+            this.handler = Optional.of((VehicleHandler) handler);
         }
     }
 
     @Override
     public @Nullable ThingHandler getThingHandler() {
-        return this.handler;
+        return handler.get();
```

Comment:
This will not work but throw a `NoSuchElementException`. You need to check for presence and return `null` if it is not present. But I must admit, that I don't see the benefit of this refactoring, checking `isPresent` is not better than checking ` != null`.

Reply:
You're right.

Reply:
Moving changes back.